src/main/java/nl/mpdev/backend_spring_boot_tech_it_easy_controller/services/TelevisionService.java

```diff
@@ -1,122 +1,145 @@
-        setFieldsIfNotNUll(existingTelevision, televisionSalesInputDto);
-        return televisionSalesMapper.toDto(televisionRepository.save(existingTelevision));
-    }
-
-
-    public void deleteTelevision(int id) {
-        Television existingTelevision = televisionRepository.findById(id).orElseThrow(() -> new RecordNotFoundException("Record not found"));
-        televisionRepository.deleteById(id);
-    }
-
-
-    // Helper class to check if one of the fields is null and therefor not being set
-    // The one that is not null can only replace the field of the existing record
-
-    private void setFieldsIfNotNUll(Object existingObject, Object incomingObject) {
-        Field[] fields = incomingObject.getClass().getDeclaredFields();
-        for (Field dtoField : fields) {
-            dtoField.setAccessible(true);
-            try {
-                Object value = dtoField.get(incomingObject);
-
-                if (value != null) {
-                    Field enityField = existingObject.getClass().getDeclaredField(dtoField.getName());
-                    enityField.setAccessible(true);
-                    enityField.set(existingObject, value);
-                }
-            } catch (IllegalAccessException e) {
-                // This is about if the fields are accesisble and are being private or not
-                throw new GeneralException("Cannot acces fields: " + dtoField.getName());
-            } catch (NoSuchFieldException e) {
-                // This is about if the field name are not the same in this case
-                throw new GeneralException("Field not found: " + dtoField.getName());
-            }
+    television.setWallBrackets(wallBracketsFound);
+    return televisionCompleteMapper.toDto(televisionRepository.save(television));
+  }
+
+  // patch
+
+  public TelevisionCompleteOutputDTO updateTelevisionFields(Long id, TelevisionCompleteInputDto televisionCompleteInputDto) {
+    Television existingTelevision = televisionRepository.findById(id).orElseThrow(() -> new RecordNotFoundException("Record not found"));
+    setFieldsIfNotNUll(existingTelevision, televisionCompleteInputDto);
+    return televisionCompleteMapper.toDto(televisionRepository.save(existingTelevision));
+  }
+
+  public void deleteTelevision(Long id) {
+    Television existingTelevision = televisionRepository.findById(id).orElseThrow(() -> new RecordNotFoundException("Record not found"));
+    televisionRepository.deleteById(id);
+  }
+
+  // Helper class to check if one of the fields is null and therefor not being set
+  // The one that is not null can only replace the field of the existing record
+
+  private void setFieldsIfNotNUll(Object existingObject, Object incomingObject) {
+    Field[] fields = incomingObject.getClass().getDeclaredFields();
+    for (Field dtoField : fields) {
+      dtoField.setAccessible(true);

```

Comment:
Hiermee zit je al je velden op public. Dat is niet iets wat je wilt doen, over het algemeen. Reflectie is een overpowered tool in Java en word meestal alleen gebruikt voor meta functies en annotaties